base/commandLine.cpp

```diff
@@ -798,20 +815,139 @@ static bool autoDetect(Common::String path) {
+static bool detectGames(Common::String path, bool addToConfig) {
+	GameList candidates = getGameList(path);
+	if (candidates.empty())
+		return false;
 
 	// Print all the candidate found
 	printf("ID                   Description\n");
 	printf("-------------------- ---------------------------------------------------------\n");
 	for (GameList::iterator v = candidates.begin(); v != candidates.end(); ++v) {
 		printf("%-20s %s\n", v->gameid().c_str(), v->description().c_str());
 	}
-	// Set the active domain to the first one to start it.
-	ConfMan.setActiveDomain(candidates.begin()->gameid());
+
+	if (addToConfig) {
+		Common::String domain = candidates[0].preferredtarget();
+		ConfMan.setActiveDomain(domain);
+	}
 	return true;
 }
 
+/** Add one of the games in the given directory, or current directory if empty */
+static bool addGame(Common::String path) {
+	GameList candidates = getGameList(path);
+	if (candidates.empty())
+		return false;
+
+	int idx = 0;
+	// Pick one if there are several games
+	if (candidates.size() > 1) {
+		// Print game list
+		printf("Several games are detected. Please pick one game to add: \n");
+		int i = 1;
+		for (GameList::iterator v = candidates.begin(); v != candidates.end(); ++i, ++v) {
+			printf("%2i. %s\n", i, v->description().c_str());
+		}
+
+		// Get user input
+		scanf("%i", &idx);
+		--idx;
+		if (idx < 0 || idx >= (int)candidates.size()) {
+			printf("Invalid index. No game added.\n");
+			return false;
+		}
+	}
+
+	Common::String domain = candidates[idx].preferredtarget();
+
+	// If game has already been added, don't add
+	if (ConfMan.hasGameDomain(domain)) {
+		printf("This game has already been added.\n");
+		return false;
+	}
+
+	// Add the name domain
+	ConfMan.addGameDomain(domain);
+
+	// Display game info
+	printf("Game Added: \n  GameID:   %s\n  Name:     %s\n  Language: %s\n  Platform: %s\n",
+			candidates[idx].gameid().c_str(),
+			candidates[idx].description().c_str(),
+			Common::getLanguageDescription(candidates[idx].language()),
+			Common::getPlatformDescription(candidates[idx].platform()));
+	return true;
+}
+
+static bool massAddGame(Common::String path) {
+	if (path.empty())
+			path = ".";
+
+	// Current directory
+	Common::FSNode startDir(path);
+	Common::Stack<Common::FSNode> scanStack;
+	scanStack.push(startDir);
+
+	// Number of games added
+	int n = 0, ndetect = 0;
+	while (!scanStack.empty()) {
+
+		Common::FSNode dir = scanStack.pop();
+		Common::FSList files;
+
+		//Collect all files from directory
+		dir.getChildren(files, Common::FSNode::kListAll);
+
+		// Get game list and add games
+		GameList candidates(EngineMan.detectGames(files));
+		if (!candidates.empty()) {
+			for (GameList::iterator v = candidates.begin(); v != candidates.end(); ++v) {
+
+				++ndetect;
+				Common::String domain = v->preferredtarget();
+				// If game was already added, don't add it
+				if (ConfMan.hasGameDomain(domain))
+					continue;
+
+				// Add game
+				ConfMan.addGameDomain(domain);
```

Comment:
Yes, I've tested it. The game can be started from launcher. I think the path has already been taken care of in the function detectGames of metaengine or somewhere else. I will take a deeper look into how it works exactly.